### PR TITLE
session: call "begin" in a transaction will auto commit, log this operation

### PR DESCRIPTION
**Pull request by tiancaiamao**
begin;
update1;
update2;
begin;    // This will auto commit previous two update operation and start a new transaction.

Log this operation maybe helpful for debugging.
@coocood @shenli

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: session/session.go [hunk 1]
```diff
@@ -956,11 +956,13 @@ func (s *session) Txn() kv.Transaction {
 
 func (s *session) NewTxn() error {
 	if s.txn.Valid() {
+		txnID := s.txn.StartTS()
 		ctx := context.TODO()
 		err := s.CommitTxn(ctx)
 		if err != nil {
 			return errors.Trace(err)
 		}
+		log.Infof("NewTxn() inside a transaction auto commit: %d", txnID)
```
Comment (zimulala) [inline session/session.go]: Could we add `connID` info to this log ?

**Comment by tiancaiamao** [comment]
PTAL @winkyao

**Comment by winkyao** [comment]
LGTM

**Pull request merged**